From 0c7daf4a2d395e04245d331c5f71d67d600716bd Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Thu, 12 Jun 2014 12:53:06 +0100
Subject: [PATCH 226/351] switch: arizona: Restart timeouts on spurious
 detection events

On suprious headphone and microphone detection events we stop the
state timeout but never restart it. If a spurious headphone event
happens during the microphone detection we might never time out.

Change-Id: I70a4886c2f3f37c780d8ef86cb6ce6a9615893d3
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/switch/switch-arizona.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/switch/switch-arizona.c b/drivers/switch/switch-arizona.c
index e60ebd5..372a102 100644
--- a/drivers/switch/switch-arizona.c
+++ b/drivers/switch/switch-arizona.c
@@ -1252,6 +1252,7 @@ static irqreturn_t arizona_hpdet_handler(int irq, void *data)
 		break;
 	default:
 		dev_warn(arizona->dev, "Spurious HPDET IRQ\n");
+		arizona_jds_start_timeout(info);
 		mutex_unlock(&info->lock);
 		return IRQ_NONE;
 	}
@@ -1291,6 +1292,7 @@ static void arizona_micd_handler(struct work_struct *work)
 		break;
 	default:
 		dev_warn(arizona->dev, "Spurious MICDET IRQ\n");
+		arizona_jds_start_timeout(info);
 		mutex_unlock(&info->lock);
 		return;
 	}
-- 
1.7.9.5

